agent_app_sample_code/00_global_config.py

```diff
@@ -0,0 +1,200 @@
+# Databricks notebook source
+# MAGIC %md
+# MAGIC # Global configuration
+# MAGIC
+# MAGIC This notebook is meant to be changed once throughout the lifetime of your Agent.
+# MAGIC
+# MAGIC At a high-level it defines the Unity Catalog locations where the agent, and associated resources, will be stored. Ideally you shouldn't have to change the values in this notebook once you've started building your agent.
+
+# COMMAND ----------
+
+# If you run this notebook directly, you need these dependencies:
+# %pip install -U -qqqq databricks-sdk mlflow mlflow-skinny
+# dbutils.library.restartPython()
+
+# COMMAND ----------
+
+# By default, will use the current user name to create a unique UC catalog/schema & vector search endpoint
+user_email = spark.sql("SELECT current_user() as username").collect()[0].username
+user_name = user_email.split("@")[0].replace(".", "").lower()[:35]
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ## Agent configuration
+# MAGIC
+# MAGIC Important: These notebooks only work on Single User clusters running DBR/MLR 14.3+.
+# MAGIC
+# MAGIC To begin with, we simply need to configure the following:
+# MAGIC 1. `AGENT_NAME`: The name of the Agent.  Used to name the app's Unity Catalog model and is prepended to the output Delta Tables + Vector Indexes
+# MAGIC 2. `UC_CATALOG` & `UC_SCHEMA`: [Create a Unity Catalog](https://docs.databricks.com/en/data-governance/unity-catalog/create-catalogs.html#create-a-catalog) and a Schema where the output Delta Tables with the parsed/chunked documents and Vector Search indexes are stored
+# MAGIC 3. `UC_MODEL_NAME`: Unity Catalog location to log and store the agent's model
+# MAGIC
+# MAGIC This notebook will also check that you are using a valid cluster type and all locations / resources exist. Any missing resources will be created.
+
+# COMMAND ----------
+
+# The name of the Agent.  This is used to name the agent's UC model and prepended to the output Delta Tables + Vector Indexes
+AGENT_NAME = "my_agent_app"
+
+# UC Catalog & Schema where outputs tables/indexs are saved
+# If this catalog/schema does not exist, you need create catalog/schema permissions.
+UC_CATALOG = f"{user_name}_catalog"
```

Comment:
I think it's more idiomatic for the catalog to be the app name - at least that's how I see people using it in practice

Reply:
Agreed, makes sense to take that as a follow-up (we can follow what e.g. MLOps stacks does)